File: algorithms/implementation/chocolate-feast/chocolate-feast-python/tests/test_chocolateFeast.py
from chocolateFeast import chocolateFeast

class TesteChocolateFeast:
    
    def testeCaso1(self):
        
        n = 10
        c = 2
        m = 5
        
        esperado = 6
        
        resultado = chocolateFeast(n, c, m)
        
        assert resultado == esperado
        
        
    def testeCaso2(self):
        
        n = 12
        c = 4
        m = 4
        
        esperado = 3
        
        resultado = chocolateFeast(n, c, m)
        
        assert resultado == esperado
        
    def testeCaso3(self):
        
        n = 6
        c = 2
        m = 2
        
        esperado = 5
        
        resultado = chocolateFeast(n, c, m)
        
        assert resultado == esperado

File: algorithms/implementation/chocolate-feast/chocolate-feast-javascript/js/testChocolateFeast.js
import {chocolateFeast} from '../js/chocolateFeast.js';

import assert from 'assert';

describe('chocolateFeast', function() {

    it('O número total de chocolates que Brenno come é: 6', function() {

        let n = 10;
        let c = 2;
        let m = 5;
        
        let esperado = 6;
        
        let resultado = chocolateFeast(n, c, m);

        assert.strictEqual(resultado, esperado)
    });
});

describe('chocolateFeast', function() {

    it('O número total de chocolates que Brenno come é: 3', function() {

        let n = 12;
        let c = 4;
        let m = 4;
        
        let esperado = 3;
        
        let resultado = chocolateFeast(n, c, m);

        assert.strictEqual(resultado, esperado)
    });
});

describe('chocolateFeast', function() {

    it('O número total de chocolates que Brenno come é: 5', function() {

        let n = 6;
        let c = 2;
        let m = 2;
        
        let esperado = 5;
        
        let resultado = chocolateFeast(n, c, m);

        assert.strictEqual(resultado, esperado)
    });
});